lib/types/common.ts

```diff
@@ -1,14 +1,34 @@
 import { I18n, TFunction } from 'next-i18next';
+import { Dispatch, SetStateAction } from 'react';
 
 export type SupportedLanguage = 'GERMAN' | 'ENGLISH';
 
-export interface FullName {
-  firstName?: string;
-  middleName?: string;
-  lastName: string;
-}
+export type IsoCode = 'SRB' | 'DEU' | 'CRO' | 'POL' | 'AUT';
 
 export interface PageProps {
   t: TFunction;
   i18n: I18n;
 }
+
+export enum ModalType {
+  NONE,
+  BASIC_INFO,
+  EXPERIENCE,
+  QUALIFICATION,
+  LANGUAGE,
+}
+
+export interface BaseEntity {
+  id: string;
+}
+export interface Address {
+  street?: string;
+  streetNo?: string;
+  city?: string;
+  postalCode?: string;
+  isoCode?: IsoCode | 'NONE';
+}
+
+export type SetValueFunction = Dispatch<
+  SetStateAction<{ id: string } & Record<string, unknown>>
```

Comment:
because we have the interface "BasEntity", can we not use it here? Like:

SetStateAction<BaseEntity & Record<string, unknown>>

Reply:
you're absolutely right. But actually, I only used the type defined here in a previous version of the selector/radio/toggle components. So, I simply deleted it.

Reply:
okay